### ISSUE TITLE
Feature: Document the withKlass option for node replacement

### ISSUE THREAD

**Issue by etrepum** [facebook/lexical#6398]
## Description

When using node replacement, it's often useful to specify the `withKlass` option so that any behaviors that were intended for the original node type should work with the replaced node type (e.g. node transforms, mutation listeners, etc.).

## Impact

Anyone replacing built-in nodes with extended versions should probably be using this, but they wouldn't know that it's even an option unless they were looking at the types or the source code. Even then, it would be tricky to know what the semantics are unless you have a pretty thorough understanding of how the editor is constructed and how listeners work.

It may even be worth a lint and/or a warning when this is not specified.

**Comment by Shopiley** [comment]
Hi, I can take this up. But just to clarify, we want to add an explanation of the `withKlass` option in the docs and also implement a lint for it?

**Comment by etrepum** [comment]
How about you just work on the documentation, adding a lint is probably a bit tricky

**Comment by etrepum** [comment]
Re-opened until we have API documentation

**Comment by etrepum** [comment]
That example should be replaced with a stackblitz example that comes from this repository like the others

**Comment by Shopiley** [comment]
Okay, do I just add the example to the repo and host it on my stackblitz account?

**Comment by etrepum** [comment]
If the example is in the repo then you can create a stackblitz URL directly to it. Take a look at any of the existing stackblitz URLs in the docs

**Comment by SansPapyrus683** [comment]
Is there a way to pass multiple possible node types to `withKlass`?

**Comment by etrepum** [comment]
No, replacement swaps the implementation from one node class to another node class. Having multiple replacement classes doesn't really make sense in that context.

**Comment by SansPapyrus683** [comment]
I have code which replaces the node with one or the other depending on the data. How would I do this while accommodating `withKlass`?

**Comment by etrepum** [comment]
You don't. Either you have one class that can handle both use cases, or you use a node transform to do the replacement instead.

**Comment by markselby9** [comment]
Happy to take this — the API documentation portion (TSDoc on `LexicalNodeReplacement`). I'll keep scope tight: TSDoc on the type and its three fields (`replace`, `with`, `withKlass`), explaining when `withKlass` matters and pointing to the runtime warning. Won't touch the stackblitz example (separate sub-task).
